x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx

```diff
@@ -84,14 +105,36 @@ export function LogRateAnalysisPanel({ slo, alert, rule }: Props) {
     getDataView();
   }, [index, dataViews, params, dataViewsService, groupBy, groupings]);
 
-  const intervalFactor = 1;
-  const alertStart = moment(alert.start);
-  const alertEnd = alert.fields[ALERT_END] ? moment(alert.fields[ALERT_END]) : undefined;
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  // We always take the "long window" of the SLO to calculate the intervalFactor.
+  const alertActionGroup =
+    alert.fields['kibana.alert.action_group'] !== 'recovered'
+      ? alert.fields['kibana.alert.action_group']
+      : ALERT_ACTION_ID;
+  const sloWindows = alert.fields['kibana.alert.rule.parameters']!.windows as WindowSchema[];
+  const relatedWindow = sloWindows.find((window) => window.actionGroup === alertActionGroup);
+  const longWindowValue = relatedWindow?.longWindow.value;
+  const longWindowUnit = relatedWindow?.longWindow.unit;
+  const lookbackDuration =
+    longWindowValue && longWindowUnit
+      ? moment.duration(longWindowValue as number, longWindowUnit as any)
+      : moment.duration(1, 'm');
+  const intervalFactor = Math.max(1, lookbackDuration.asSeconds() / 60);
 
-  const timeRange = {
-    min: alertStart.clone().subtract(15 * intervalFactor, 'minutes'),
-    max: alertEnd ? alertEnd.clone().add(1 * intervalFactor, 'minutes') : moment(new Date()),
-  };
+  const actionGroup = getActionGroupFromReason(alert.reason);
+  const actionGroupWindow = (
```

Comment:
@walterra Can't we use here the `alertActionGroup` we define in line 112 above?

Reply:
The difference I can see is that `alertActionGroup` above fallbacks to `ALERT_ACTION_ID`, where as the `alertActionGroup` fallbacks to whatever the `getActionGroupFromReason` fallbacks with is `LOW_PRIORITY_ACTION_ID`

Reply:
The way I understand the alert metadata in line `112` we just get the name like `slo.burnRate.alert`, here we then get the actual metadata of a window including the values for short and long window.

Reply:
Sorry I meant to say, can't we use `const relatedWindow = sloWindows.find((window) => window.actionGroup === alertActionGroup);` that I define above? You do pretty much the same for the `actionGroupWindow`